simpleviewinc/keg-cli
# Move src/lib/git to repos/git-lib

> This PR is branched off the changes from [this pr](https://github.com/simpleviewinc/keg-cli/pull/64)
> Please do the other PR first before doing this one

## Context

* Continue the migration of the keg-cli to a mono-repo

## Goal

* Migrate `src/libs/git` to `repos/git-lib`
* Make `git-lib` it's own package that can be used independently from the `keg-cli`

## Updates

* Moved `src/libs/git` to `repos/git-lib`
* Moved a number of `utils/git` methods to `repos/git-lib/src/utils`
* Added separate package.json for `git-lib`

## Testing

**Setup**
* Pull this PR => `keg cli && keg pr 66`
* Install deps for git-lib repo => `cd repos/git-lib && yarn install`

**Tests**
* Run the `keg-cli` git tasks to ensure they still work as expected. Run each command in sequence =>
  * `keg fetch`
    * Ensure it fetches changes from github
  * `keg pull`
    * Ensure it tries to pull changes from github
  * `keg br`
    * Ensure you see a list of all branches
  * `keg br -n my-test-branch`
    * Edit some file in the keg-cli, and commit the change
  * `keg push`
    * Ensure it pushes to github  
  * `keg br move-git-to-utils`
    * Ensure it switches back to this branch
  * `keg br -d my-test-branch`
    * This deletes the branch locally. Please also manually delete it in github 
  * `keg log`
    * Ensure the  the git commit history is displayed
